src/dftracer/core/utils/configuration_manager.cpp

```diff
@@ -291,12 +307,17 @@ dftracer::ConfigurationManager::ConfigurationManager()
     }
     DFTRACER_LOG_DEBUG("ENV ConfigurationManager.tids %d", this->tids);
     const char *env_enable_aggregation = getenv(DFTRACER_ENABLE_AGGREGATION);
-    if (env_enable_aggregation != nullptr &&
-        strcmp(env_enable_aggregation, "1") == 0) {
-      this->enable_aggregation = true;
+    if (env_enable_aggregation != nullptr) {
+      convert(env_enable_aggregation, this->aggregation_type);
+    }
+    DFTRACER_LOG_DEBUG("ENV ConfigurationManager.enable_aggregation %s",
+                       to_string(this->aggregation_type).c_str());
```

Comment:
Wonder if to_string could just be "to_char" returning a char * to avoid unnecessary const char * -> string -> char * conversion. Minor nit.

Reply:
problem with char* directly is freeing of memory

I hate multiple new and free. They are recipe for disaster.

This is happening at init and for just simple conversion I prefer using std::string.